controllers/cloudstackcluster_controller_test.go

```diff
@@ -77,6 +77,8 @@ var _ = Describe("CloudStackClusterReconciler", func() {
 
 	It("Should create a cluster", func() {
 		By("Fetching a CS Cluster Object")
+		CS.EXPECT().GetOrCreateCluster(gomock.Any()).MinTimes(1)
```

Comment:
This may actually have been our random test failure issue.

I think I was just wrong...

I ran the test locally and saw a failure after only ~15 seconds. So, the failure wasn't timeout based.

I then realize it was actually complaining about a method call that happened -- not a missing call.

Turns out the call can sometimes happen early.